stdlib/string.gr

```diff
@@ -1867,30 +1867,49 @@ export let forEachCodePointi = (fn: (Number, Number) -> Void, str: String) => {
   void
 }
 
+@unsafe
 let trimString = (str: String, end: Bool) => {
-  let chars = explode(str), charsLength = length(str)
-  let mut i = 0, offset = 1
+  let (+) = WasmI32.add
+  let (-) = WasmI32.sub
+  let (<) = WasmI32.ltU
+  let (!=) = WasmI32.ne
+
+  let mut stringPtr = WasmI32.fromGrain(str)
+  let charsLength = WasmI32.load(stringPtr, 4n)
+  stringPtr += 8n
+  let mut i = 0n, offset = 1n
   if (end) {
-    i = charsLength - 1
-    offset = -1
+    i = charsLength - 1n
+    offset = -1n
   }
-  for (; i < charsLength && i > -1; i += offset) {
-    let currentChar = chars[i]
+  let mut count = 0n
+  for (; i < charsLength; i += offset) {
+    // Get The Character
+    let byte = WasmI32.load8U(stringPtr, i)
```

Comment:
It's important to note that since the string is UTF-8, this byte may not necessarily be a full code point.

Reply:
Can you update the comment to reflect this?

Reply:
Grain definitely uses UTF-8, sir! I meant that one byte may not be a full codepoint because UTF-8 codepoints span between 1 and 4 bytes. Could you update the comment again?

Reply:
`Get The Byte, May not be a full codepoint as grain uses utf-8, a codepoint may be 1-4 bytes`, is this wording preferred I didn't mean to write `utf-16` oops